screeps-game-api/src/objects/mod.rs

```diff
@@ -190,16 +190,22 @@ impl HasPosition for RoomPosition {
     }
 }
 
-/// All RoomObjects have positions.
-impl<T> HasPosition for T
-where
-    T: RoomObjectProperties,
-{
-    fn pos(&self) -> RoomPosition {
-        js_unwrap!(@{self.as_ref()}.pos)
+/// Trait covering all objects with an id.
+pub trait HasId : AsRef<Reference> {
```

Comment:
I think this also deserves a `pub use` in `prelude` since it's a trait implementing functionality, and we want all regular methods on objects to be available as long as someone imports `screeps::prelude::*`.